ionoscloud/data_source_dataplatform_cluster.go

```diff
@@ -0,0 +1,373 @@
+package ionoscloud
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	dataplatform "github.com/ionos-cloud/sdk-go-dataplatform"
+	dataplatformService "github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/dataplatform"
+	"gopkg.in/yaml.v3"
+	"log"
+	"regexp"
+	"strings"
+)
+
+func dataSourceDataplatformCluster() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDataplatformClusterRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:         schema.TypeString,
+				Description:  "The id of your cluster.",
+				Optional:     true,
+				ValidateFunc: validation.All(validation.StringMatch(regexp.MustCompile("^[A-Za-z0-9][-A-Za-z0-9_.]*[A-Za-z0-9]$"), "")),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of your cluster.",
+				Optional:    true,
+			},
+			"partial_match": {
+				Type:        schema.TypeBool,
+				Description: "Whether partial matching is allowed or not when using name argument.",
+				Default:     false,
+				Optional:    true,
+			},
+			"data_platform_version": {
+				Type:        schema.TypeString,
+				Description: "The version of the Data Platform.",
+				Computed:    true,
+			},
+			"datacenter_id": {
+				Type:        schema.TypeString,
+				Description: "The UUID of the virtual data center (VDC) the cluster is provisioned.",
+				Computed:    true,
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				Description: "Starting time of a weekly 4 hour-long window, during which maintenance might occur in hh:mm:ss format",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:        schema.TypeString,
+							Description: "Time at which the maintenance should start. Must conform to the 'HH:MM:SS' 24-hour format.",
+							Computed:    true,
+						},
+						"day_of_the_week": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"config": {
+				Type:      schema.TypeList,
+				Computed:  true,
+				Sensitive: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"api_version": {
+							Type:      schema.TypeString,
+							Computed:  true,
+							Sensitive: true,
+						},
+						"current_context": {
+							Type:      schema.TypeString,
+							Computed:  true,
+							Sensitive: true,
+						},
+						"kind": {
```

Comment:
@cristiGuranIonos not sure about this: for other fields we have `Sensitive: true`. For `kind`, we don't. Is this intended or should be `kind` also be treated like sensitive?

Reply:
it's only the type of k8s object